src/React/Widgets/AnnotationStoreEditorWidget/index.js

```diff
@@ -41,7 +42,8 @@ export default function annotationStoreEditorWidget(props) {
 
   if (props.annotation && props.annotations[props.annotation.id]) {
     const storedSelectedAnnotation = props.annotations[props.annotation.id];
-    if (storedSelectedAnnotation.generation === props.annotation.generation) {
+    if (storedSelectedAnnotation.generation === props.annotation.generation ||
+        deepEquals(Object.assign({}, storedSelectedAnnotation, { generation: props.annotation.generation }), props.annotation)) {
```

Comment:
It doesn't make much difference - right now, AnnotationBuilder will increment the generation whenever you mutate an annotation, even if you set it back to the same thing. So you'd have to ignore AnnotationBuilder if you wanted to change the annotation and have AnnotationStoreEditorWidget treat it as unmodified. I'll push the second way...